packages/wonder-blocks-dropdown/components/dropdown.test.js

```diff
@@ -0,0 +1,241 @@
+        this.setState({
+            selectedValues: update,
+        });
+
+    render() {
+        const dropdownItems = [
+            <OptionItem label="A" value="A" />,
+            <OptionItem label="B" value="B" />,
+            <OptionItem label="C" value="C" />,
+            <OptionItem label="D" value="D" />,
+        ];
+
+        return (
+            <Dropdown
+                menuItems={dropdownItems}
+                onChange={this.handleChange}
+                selectedValues={this.state.selectedValues}
+            >
+                {(state) => <h1>Manage students</h1>}
+            </Dropdown>
+        );
+    }
+}
+
+describe("Dropdown Launcher", () => {
+    window.scrollTo = jest.fn();
+    window.getComputedStyle = jest.fn();
+    let dropdown;
+
+    beforeEach(() => {
+        dropdown = mount(
+            <Dropdown
+                testId="teacher-menu"
+                selectedValues={["A"]}
+                menuItems={[
+                    <OptionItem label="A" value="A" />,
+                    <OptionItem label="B" value="B" />,
+                    <OptionItem label="C" value="C" />,
+                    <ActionItem
+                        label="Profile"
+                        href="http://khanacademy.org/profile"
+                        testId="profile"
+                    />,
+                    <ActionItem
+                        label="Teacher dashboard"
+                        href="http://khanacademy.org/coach/dashboard"
+                        testId="dashboard"
+                    />,
+                    <ActionItem
+                        label="Settings (onClick)"
+                        onClick={() => {}}
+                        testId="settings"
+                    />,
+                    <ActionItem
+                        label="Help"
+                        disabled={true}
+                        onClick={() => {}}
+                        testId="help"
+                    />,
+                    <ActionItem
+                        label="Feedback"
+                        disabled={true}
+                        href="/feedback"
+                        testId="feedback"
+                    />,
+                ]}
+            >
+                {(state) => (
+                    <IconButton icon={icons.search} aria-label="search" />
+                )}
+            </Dropdown>,
+        );
+    });
+
+    afterEach(() => {
+        unmountAll();
+    });
+
+    it("opens when the anchor is clicked", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+
+        // Act
+        anchor.simulate("click");
+
+        // Assert
+        expect(dropdown.state("opened")).toBe(true);
+    });
+
+    it("can handle keyboard navigation", () => {
+        // Arrange
+        const anchor = dropdown.find(IconButton);
+        anchor.simulate("click");
+        const dropdownCore = dropdown.find(DropdownCore);
+
+        // Act & Assert
```

Comment:
This should really be split into separate test cases, e.g.
- `it("should navigate to the first item when pressing down", () => {...})`
- `it("should navigate from one item to the next when pressing down", () => {...})`

By splitting these up, you'll be able to have separate `Act` and `Assert` blocks.